asynciterator.ts

```diff
@@ -472,6 +472,23 @@ export class AsyncIterator<T> extends EventEmitter {
     return this.transform({ filter: self ? filter.bind(self) : filter });
   }
 
+  /**
+   * Returns a new iterator containing all of the unique items in the original iterator.
+   * @param by The property by which to determine uniqueness (e.g. stringification). This is the identity
+   * by default.
+   * @returns An iterator with duplicates filtered out.
+   */
+  uniq(by: (item: T) => any = item => item): AsyncIterator<T> {
```

Comment:
I think the signature for `by` should read `(item: T) => string` to prevent serialization mistakes by passing functions that may lead to collisions due to implicit `toString()` calls resulting in hashes similar to `[Object object]`.

Reply:
Oh nevermind, I just realized `Set` works any kind of type. Silly me.

Reply:
Oh nevermind, I just realized `Set` works any kind of type. Silly me.